### PR TITLE
Update TestProwJobToPod unit test

### PR DESCRIPTION
**Pull request by kargakis**

### CODE REVIEW

File: prow/pjutil/pjutil_test.go [hunk 1]
```diff
@@ -114,71 +115,8 @@ func TestProwJobToPod(t *testing.T) {
 		if err != nil {
 			t.Errorf("unexpected error: %v", err)
 		}
-		// TODO: For now I am just comparing fields manually, eventually we
-		// should port the semantic.DeepEqual helper from the api-machinery
```
Comment (BenTheElder) [inline prow/pjutil/pjutil_test.go]: :tada:

**Comment by BenTheElder** [comment]
code LGTM, imports changed so `hack/update-bazel.sh`  ¯\\\_(ツ)\_/¯

**Comment by BenTheElder** [comment]
hmm looks like a network flake
/retest

**Comment by BenTheElder** [comment]
:this_is_fine:
/retest

**Comment by BenTheElder** [comment]
/lgtm cancel

OK, this is not what we have vendored, we have equality.Semantic like the comment, not Deep Equal

https://github.com/kubernetes/test-infra/blob/a6ec629b471b5a042d1a5fb58d9530adf8ac3eff/vendor/k8s.io/apimachinery/pkg/api/equality/semantic.go#L29

File: prow/pjutil/pjutil_test.go [hunk 2]
```diff
@@ -114,71 +115,8 @@ func TestProwJobToPod(t *testing.T) {
...
-		}
-		if dumpGotEnv {
-			t.Errorf("expected env:\n%#v\ngot:\n%#v\n", expectedContainer.Env, gotContainer.Env)
-		}
-		if expectedContainer.Image != gotContainer.Image {
-			t.Errorf("expected image: %s, got: %s", expectedContainer.Image, gotContainer.Image)
-		}
-		if test.expected.Spec.RestartPolicy != got.Spec.RestartPolicy {
-			t.Errorf("expected restart policy: %s, got: %s", test.expected.Spec.RestartPolicy, got.Spec.RestartPolicy)
+		if !equality.DeepEqual(got, test.expected) {
```
Comment (BenTheElder) [inline prow/pjutil/pjutil_test.go]: this doesn't match what we have vendored, not sure about upstream

Comment (kargakis) [reply prow/pjutil/pjutil_test.go]: fixed

The test also needs fixing

/hold

**Comment by stevekuznetsov** [comment]
Looks like the new equality function expects the maps to be sorted?

**Comment by stevekuznetsov** [comment]
Err, in the same "order" ?

**Pull request closed without merging**